doc/overviews/helm-charts-release-process.md

```diff
@@ -0,0 +1,318 @@
+
+    Note over Dev, ArtifactHub: 🔄 Phase 4: Helm Charts Repository Processing
+    ChartRepo->>SyncChart: Trigger sync-chart-created.yaml workflow
+    SyncChart->>GitHub: Download chart from release asset URL
+    SyncChart->>SyncChart: Update charts/index.yaml<br/>Add new chart version
+    SyncChart->>ChartRepo: Commit updated index to main branch
+    ChartRepo->>SyncIndex: Trigger sync-index.yaml workflow<br/>(on charts/index.yaml change)
+    SyncIndex->>SyncIndex: Copy index.yaml to gh-pages branch
+    SyncIndex->>ChartRepo: Publish to kuadrant.io/helm-charts
+
+    Note over Dev, ArtifactHub: 🎯 Phase 5: Artifact Hub Publishing
+    ChartRepo->>ArtifactHub: Updated helm repository detected
+    ArtifactHub->>ArtifactHub: Auto-discovery scan of repository
+    ArtifactHub->>ArtifactHub: Chart available at artifacthub.io
+```
+
+### Detailed Release Process
+
+#### Phase 1: Release Preparation
+
+**Trigger**: Developer manually runs the [Automated Release Workflow](../../.github/workflows/automated-release.yaml)
+
+**Workflow**: [automated-release.yaml](../../.github/workflows/automated-release.yaml)
+
+**Automated Steps**:
+1. **Updates Versions**: The workflow updates the `release.yaml` file with all component versions:
+    - Kuadrant Operator version
+    - Authorino Operator version
+    - Limitador Operator version
+    - DNS Operator version
+    - WASM Shim version
+    - Console Plugin version
+
+2. **Runs Prepare Release**: Executes `make prepare-release` which:
+    - Runs the [release script](../../utils/release/release.sh) in phases:
+        - `pre-validation` - Validates all dependencies are available
+        - `dependencies` - Processes and validates dependency versions
+        - `operator` - Builds operator manifests and helm charts
+        - `post-validation` - Final validations
+    - **Builds Helm Charts** via `make_chart.sh` [script](../../utils/release/operator/make_chart.sh) which:
+        - Sets the desired WASM shim image in manager deployment
+        - Sets the operator image reference
+        - Builds helm chart templates from kustomize manifests (`kustomize build config/helm`)
+        - Updates Chart.yaml with version and dependency versions
+
+3. **Creates Pull Request**:
+    - Creates base branch `release-v{MAJOR.MINOR}` (e.g., `release-v1.0`) if it doesn't exist
+    - Uses `peter-evans/create-pull-request` action which automatically:
+      - Creates working branch `release-v{FULL_VERSION}` (e.g., `release-v1.0.0`)
+      - Commits all changes from `make prepare-release`
+      - Opens PR from working branch to base branch
+
+#### Phase 2: Release Creation
+
+**Overview**: Once the release PR is ready, developers can choose between two approaches for creating the actual release:
+
+1. **Automated Process** (Recommended): The release is created automatically when the PR is merged
+2. **Manual Process**: Follow the detailed steps in [RELEASE.md](../../RELEASE.md)
```

Comment:
I would hoist this to the _Phase 1: Release Preparation_, phrasing it as:

```md
**Trigger**: Developer could manually run the [Automated Release Workflow](../../.github/workflows/automated-release.yaml) or follow the detailed steps in [RELEASE.md](../../RELEASE.md) in order to do the release process manually.
```

Reply:
updated  [here](https://github.com/Kuadrant/kuadrant-operator/compare/9c01b70e8b8ad5f643461bf4d3d8517aa32ea00b..e6e14c7bf7731a6a0d7c3292a284472978ed6781)